alf/metrics/metrics.py

```diff
@@ -0,0 +1,251 @@
+# Copyright (c) 2020 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A set of metrics.
+Converted to PyTorch from the TF version.
+https://github.com/tensorflow/agents/blob/master/tf_agents/metrics/tf_metrics.py
+"""
+
+from alf.metrics import metric
+import torch
+
+
+class THDeque(torch.nn.Module):
+    """Torch (TH) deque backed by torch.nn.Parameter storage."""
+
+    def __init__(self, max_len, dtype):
+        """Constructor.
+
+        Args:
+            max_len (int): maximum length of the deque
+            dtype (torch.dtype): dtype of the content of the deque
+        """
+        super().__init__()
+        shape = (max_len, )
+        self.dtype = dtype
+        self._max_len = torch.nn.Parameter(
```

Comment:
It's better to use register_buffer(), otherwise this may be picked up by optimizers. See https://pytorch.org/docs/stable/nn.html#torch.nn.Module.register_buffer